lib/Sema/CSApply.cpp

```diff
@@ -5121,6 +5121,11 @@ namespace {
       return E;
     }
 
+    Expr *visitTypeJoinExpr(TypeJoinExpr *E) {
+      // Type-join expressions should not appear in the type-checked AST.
```

Comment:
It is possible have a `TypeJoinExpr` here due to a bug? If it is maybe is worth assert or `llvm_unreachable` so it is noticeable?

Reply:
Yes, thanks for spotting this! Not having llvm_unreachable here is left-over from initial changes that introduced TypeJoinExpr

Reply:
Fixed!